src/EditorFeatures/CSharpTest/Completion/CompletionProviders/OverrideCompletionProviderTests.cs

```diff
@@ -2693,5 +2694,60 @@ public class Class1
                 Assert.False(completionList.Items.Any(c => c.DisplayText == "Goo()"));
             }
         }
+
+        // Override completion provider tests that use a workspace with an assembly reference
+        [WorkItem(24490, "https://github.com/dotnet/roslyn/issues/24490")]
+        public class OverrideCompletionProviderWithAssemblyReferenceTests : OverrideCompletionProviderTests
+        {
+            public OverrideCompletionProviderWithAssemblyReferenceTests(CSharpTestWorkspaceFixture workspaceFixture) : base(workspaceFixture)
+            {
+            }
+
+            private TestWorkspace MakeWorkspaceWithAssemblyReference(string fileContents)
+            {
+                var options = CSharpParseOptions.Default.WithLanguageVersion(LanguageVersion.Latest);
+                var workspace = TestWorkspace.CreateCSharp(fileContents, parseOptions: options);
+
+                var markup = @"
+public abstract class Base
+{
+    public abstract void M(in int x);
+}";
+
+                var compilation = workspace.CurrentSolution.Projects.Single().GetCompilationAsync().GetAwaiter().GetResult();
+                var libComp = compilation.RemoveAllSyntaxTrees().AddSyntaxTrees(CSharpSyntaxTree.ParseText(markup, options: options));
+                var libRef = MetadataReference.CreateFromImage(Test.Utilities.CompilationExtensions.EmitToArray(libComp));
+
+                var project = workspace.CurrentSolution.Projects.Single();
+                var updatedProject = project.AddMetadataReference(libRef);
```

Comment:
I think we have support for doing all this in the workspace already...? If not, might be better to add there.

Reply:
This is following the model of the previous test.

---
In reply to: [164599566](https://github.com/dotnet/roslyn/pull/24492#discussion_r164599566) [](ancestors = 164599566)